marconi-chain-index/src/Marconi/ChainIndex/Indexers/MintBurn.hs

```diff
@@ -292,35 +287,58 @@ type instance RI.StorableMonad MintBurnHandle = IO
 
 newtype instance RI.StorableEvent MintBurnHandle
   = MintBurnEvent TxMintEvent
+  deriving (Show)
 
 data instance RI.StorableQuery MintBurnHandle
-  = ByTxIdAndPolicyId C.TxId C.PolicyId
-  | Everything
+  = QueryByAssetId C.PolicyId C.AssetName (Maybe C.SlotNo)
+  | QueryAllMintBurn (Maybe C.SlotNo)
+  deriving (Show)
 
 newtype instance RI.StorableResult MintBurnHandle
   = MintBurnResult [TxMintRow]
+  deriving (Show)
 
 instance RI.Queryable MintBurnHandle where
-  queryStorage queryInterval memoryEvents (MintBurnHandle sqlCon _k) query = case query of
-    Everything                      -> toResult <$> sqliteSelectAll sqlCon interval
-    ByTxIdAndPolicyId txId policyId -> toResult <$> sqliteSelectByTxIdPolicyId sqlCon interval txId policyId
-    where
-      toResult storedEvents = MintBurnResult $ do
-        TxMintEvent slotNo blockHeaderHash txAssets <- storedEvents <> map coerce (toList memoryEvents)
-        (txId, mintAssets) <- NE.toList txAssets
-        MintAsset policyId assetName quantity redeemerIx redeemerData <- NE.toList mintAssets
-        pure $
-            TxMintRow
-                slotNo
-                blockHeaderHash
-                txId
-                policyId
-                assetName
-                quantity
-                redeemerIx
-                redeemerData
-
-      interval = intervalToWhereClause queryInterval
+  queryStorage queryInterval memoryEvents (MintBurnHandle sqlCon _k) query =
+      case query of
+        QueryAllMintBurn Nothing -> toResult Nothing <$> queryStoredTxMintEvents sqlCon interval
+        QueryAllMintBurn (Just slotNo) -> do
+            let conditions = interval <> (["slotNo <= :slotNo"], [":slotNo" := slotNo])
+            toResult (Just slotNo) <$> queryStoredTxMintEvents sqlCon conditions
+        QueryByAssetId policyId assetName Nothing -> do
```

Comment:
I think this and the following case can be joined by just adding the slotNo element to the conditions list?

Reply:
Good idea

Reply:
d32ea77fe3de6f145417fa5588b9d82b41760fda